src/runtime_src/core/edge/drm/zocl/zocl_ioctl.c

```diff
@@ -318,6 +319,14 @@ int zocl_pcap_download_ioctl(struct drm_device *dev, void *data,
 	uint64_t primary_fw_len;
 	struct axlf_section_header *primaryHeader;
 
+	 /* On Non PR platform, it shouldn't never go to this point.
+	  * On PR platform, the fpga_mgr should be alive.
+	 */
+	if (!zdev->fpga_mgr) {
```

Comment:
It is called from 3 places now zocl_pcap_download, zocl_load_pdi and zocl_read_axlf_ioctl.

Reply:
I will add this check for all of them. Thanks

Reply:
LGTM so far